Dockerfile

```diff
@@ -8,8 +8,10 @@ ADD . /workdir
 
 RUN apt-get update && apt-get install -y software-properties-common
 
-RUN curl -fsSL https://apt.releases.hashicorp.com/gpg | apt-key add -
-RUN apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main"
-RUN apt-get update && apt-get install -y terraform
+RUN wget -O- https://apt.releases.hashicorp.com/gpg | gpg --dearmor -o /usr/share/keyrings/hashicorp-archive-keyring.gpg
+RUN echo "deb [signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" | tee /etc/apt/sources.list.d/hashicorp.list
+RUN apt update && apt install -y terraform
 
-ENTRYPOINT ["/workdir/scripts/tooling.py"]
\ No newline at end of file
+RUN git config --global --add safe.directory /workdir
+
+ENTRYPOINT ["/workdir/scripts/tooling.py"]
```

Comment:
The buildkite task was failing, so I had to implement some fixes. Specifically:

* The `git config ...` line was added to prevent [this error](https://buildkite.com/wellcomecollection/terraform-module-terraform-aws-sqs/builds/23#0191c717-02aa-4675-adf0-579e55d49c0c/166-170)
* The other changes were made to prevent [this error](https://buildkite.com/wellcomecollection/terraform-module-terraform-aws-sqs/builds/15#0191c6dd-e8cb-4f4f-8690-8bdb6dedf4c4/115-142). The changes follow [the official Terraform instructions](https://developer.hashicorp.com/terraform/install)